### ISSUE TITLE
Suggested tags / auto-tagging untagged items

### ISSUE THREAD

**Issue by elazar** [astralapp/astral#54]
This is a very rough idea / proposal, but I'm filing this issue to get discussion going. I'd be willing to help with implementation.

Essentially, if your initial import is large (mine is bordering on 1K), tagging can be a laborious and tedious process. It would be cool if there was a way to have Astral help the user with that.

This could take the form of searching repo descriptions and README files for existing tags or running that data through something like [OpenCalais](http://www.opencalais.com/), and ultimately presenting the results to the user for approval and applying approved tags to relevant items.

Potentially related: #9 #14

**Comment by syropian** [comment]
Yeah, having some sort of quasi-intelligent auto-tagging has kind of been a thing floating around the app for a while. You can already auto-tag by language (in settings), and I also attempted a "crowd-sourced" auto-tagging mechanism, where for each of your starred repos, it would look for that same repo across other users, and see what they starred it with. If a repo had been tagged at least (5?) times with a common tag across all users, it would add that tag to yours too.

In theory it was a neat solution, in practice, it didn't scale well, because the # of hits to the DB was causing a crazy amount of strain on the server. There's likely other solutions I could use - it's just a matter of making sure the user experience is smooth and ensuring that I don't need to sign up for some service I can't afford.

That being said, it's definitely something I want to pursue further, and hopefully with Astral being an open source project, more intelligent solutions than my own will come about 😅

**Comment by muescha** [comment]
there should be a suggested list of tags which i can check (and it should be sorted by count)
but you dont need a big database for this. just save on every new tag assignment the new `taglist = [(tag:count)]` as json per repo - you can then load this json easy :)

**Comment by aminya** [comment]
Many GitHub repositories already have tags, it would be nice to suggest those in the suggestion list.

![image](https://user-images.githubusercontent.com/16418197/110041040-4d4cf980-7d09-11eb-964b-bf767034168c.png)

**Comment by a67793581** [comment]
Has there been any progress on this feature? I'm eager to use it as well!
